dit/dit.py

```diff
@@ -424,47 +439,52 @@ def _id_parse(self, argv):
 
     def _gname_parse(self, argv):
         names = argv.pop(0).split(self.separator)
+        if len(names) > 3:
+            raise Exception("Invalid gname format")
+        names = [name if name != self.root_name_cmd else self.root_name for name in names]
         names = names + [None] * (3 - len(names))
         group, subgroup, task = names
 
-        if group is None:
-            group = self.current_group
-        elif group == "":
-            group = None
-
-        if subgroup is None:
-            subgroup = self.current_subgroup
-        elif subgroup == "":
-            subgroup = None
-
         for name in [group, subgroup, task]:
-            if not self._is_valid_name(name):
-                raise Exception("Invalid name %s" % name)
+            if name and not self._is_valid_name(name):
+                raise Exception("Invalid name: %s" % name)
 
         return (group, subgroup, task)
 
     def _name_parse(self, argv):
         names = argv.pop(0).split(self.separator)
+        if len(names) > 3:
+            raise Exception("Invalid name format")
+        names = [name if name != self.root_name_cmd else self.root_name for name in names]
         names = [None] * (3 - len(names)) + names
+
         group, subgroup, task = names
 
         if group is None:
-            group = self.current_group
-        elif group == "":
-            group = None
+            if self.current_group is not None:
+                group = self.current_group
```

Comment:
`None` should be meaningful and semantically different from `""`.

The bug was caused because we were using `None` instead of `""` in the first place.

The function should be able to return `None` to indicate the absence of a value, and `""` to indicate the root.